src/main/java/org/shanerx/tradeshop/objects/ListManager.java

```diff
@@ -0,0 +1,126 @@
+ * any damages of any kind, whether be material or moral, which were
+ * caused by their contribution(s) to the project. See the full License for more information
+ */
+
+package org.shanerx.tradeshop.objects;
+
+import org.bukkit.Material;
+import org.bukkit.block.BlockFace;
+import org.shanerx.tradeshop.enumys.Setting;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+@SuppressWarnings("unused")
+public class ListManager {
+
+	private ArrayList<String> blacklist = new ArrayList<>();
+	private ArrayList<BlockFace> directions = new ArrayList<>();
+	private ArrayList<Material> inventories = new ArrayList<>();
+
+
+	public ListManager() {
+		reload();
+	}
+
+	public ArrayList<BlockFace> getDirections() {
+		return directions;
+	}
+
+	public ArrayList<Material> getInventories() {
+		return inventories;
+	}
+
+	public ArrayList<String> getBlacklist() {
+		return blacklist;
+	}
+
+	public boolean isBlacklisted(Material mat) {
+        return blacklist.contains(mat.toString());
+    }
+
+	public boolean isDirection(BlockFace face) {
+		return directions.contains(face);
+	}
+
+	public boolean isInventory(Material mat) {
+		return inventories.contains(mat);
+	}
+
+
+	public void reload() {
+		updateBlacklist();
+		updateDirections();
+		updateMaterials();
+	}
+
+    public void clearManager() {
+        inventories.clear();
+        blacklist.clear();
+		directions.clear();
+	}
+
+	private void updateBlacklist() {
+		blacklist.clear();
+		blacklist.addAll(Setting.ILLEGAL_ITEMS.getStringList().stream().map(String::toLowerCase).collect(Collectors.toList()));
+	}
+
+	private void updateDirections() {
+		directions.clear();
+		ArrayList<BlockFace> allowed = new ArrayList<>(Arrays.asList(BlockFace.DOWN, BlockFace.WEST, BlockFace.SOUTH, BlockFace.EAST, BlockFace.NORTH, BlockFace.UP));
+
+		for (String str : Setting.ALLOWED_DIRECTIONS.getStringList()) {
+			if (allowed.contains(BlockFace.valueOf(str)))
+				directions.add(BlockFace.valueOf(str));
+		}
+	}
+
+	private void updateMaterials() {
+		inventories.clear();
+		ArrayList<Material> allowed = new ArrayList<>(Arrays.asList(Material.CHEST, Material.TRAPPED_CHEST, Material.DROPPER, Material.HOPPER, Material.DISPENSER));
+
+		for (String str : Setting.ALLOWED_SHOPS.getStringList()) {
+			if (str.equalsIgnoreCase("shulker")) {
+				inventories.addAll(Arrays.asList(Material.SHULKER_BOX,
+						Material.WHITE_SHULKER_BOX,
+						Material.ORANGE_SHULKER_BOX,
+						Material.MAGENTA_SHULKER_BOX,
+						Material.LIGHT_BLUE_SHULKER_BOX,
+						Material.YELLOW_SHULKER_BOX,
+						Material.LIME_SHULKER_BOX,
+						Material.PINK_SHULKER_BOX,
+						Material.GRAY_SHULKER_BOX,
+						Material.LIGHT_GRAY_SHULKER_BOX,
+						Material.CYAN_SHULKER_BOX,
+						Material.PURPLE_SHULKER_BOX,
+						Material.BLUE_SHULKER_BOX,
+						Material.BROWN_SHULKER_BOX,
+						Material.GREEN_SHULKER_BOX,
+						Material.RED_SHULKER_BOX,
+						Material.BLACK_SHULKER_BOX));
```

Comment:
There are better ways to do this, but fine. We probably discussed this after the release of 1.9.

Reply:
I'm doing it this way so that each shulker type doesn't needed to be typed into the config. Although i guess making an array for that is useless and i should just call #add for each one. *changed

Reply:
I am aware of that, but still there are better ways of determining whether the block is some kind of shulker box which do not involve a dozen enum comparisons.

Reply:
So all this piece is doing is adding the shulker materials to the list that also contains all other valid storage types for the shops. and checking materials is the easiest way since im not just checking for shulkers.